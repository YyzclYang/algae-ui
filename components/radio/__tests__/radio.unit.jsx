import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Radio from '../index';

describe('Radio', () => {
  it('渲染一个 Radio', () => {
    const component = TestRenderer.create(<Radio>radio</Radio>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 checked', () => {
    const component = TestRenderer.create(
      <Radio checked>radio</Radio>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 disabled', () => {
    const component = TestRenderer.create(
      <Radio disabled>radio</Radio>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('不传递子节点', () => {
    const component = TestRenderer.create(<Radio disabled />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('响应 onChange 事件', () => {
    const fn = jest.fn();
    const component = mount(<Radio onChange={fn}>radio</Radio>);
    component.find('input').simulate('change');
    expect(fn).toBeCalled();
  });

  it('disabled 时不可响应 onChange 事件', () => {
    const fn = jest.fn();
    const component = mount(
      <Radio disabled onChange={fn}>
        radio
      </Radio>
    );
    component.find('input').simulate('change');
    expect(fn).not.toBeCalled();
  });
});
