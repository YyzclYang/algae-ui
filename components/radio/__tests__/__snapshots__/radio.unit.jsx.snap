// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio 不传递子节点 1`] = `
<label
  className="algae-ui-radio-wrapper algae-ui-radio-disabled"
>
  <span
    className="algae-ui-radio"
  >
    <input
      checked={false}
      className="algae-ui-radio-input"
      disabled={true}
      onChange={[Function]}
      type="radio"
    />
    <span
      className="algae-ui-radio-inner"
    />
  </span>
</label>
`;

exports[`Radio 接受 checked 1`] = `
<label
  className="algae-ui-radio-wrapper algae-ui-radio-checked"
>
  <span
    className="algae-ui-radio"
  >
    <input
      checked={true}
      className="algae-ui-radio-input"
      disabled={false}
      onChange={[Function]}
      type="radio"
    />
    <span
      className="algae-ui-radio-inner"
    />
  </span>
  <span
    className="algae-ui-radio-text"
  >
    radio
  </span>
</label>
`;

exports[`Radio 接受 disabled 1`] = `
<label
  className="algae-ui-radio-wrapper algae-ui-radio-disabled"
>
  <span
    className="algae-ui-radio"
  >
    <input
      checked={false}
      className="algae-ui-radio-input"
      disabled={true}
      onChange={[Function]}
      type="radio"
    />
    <span
      className="algae-ui-radio-inner"
    />
  </span>
  <span
    className="algae-ui-radio-text"
  >
    radio
  </span>
</label>
`;

exports[`Radio 渲染一个 Radio 1`] = `
<label
  className="algae-ui-radio-wrapper"
>
  <span
    className="algae-ui-radio"
  >
    <input
      checked={false}
      className="algae-ui-radio-input"
      disabled={false}
      onChange={[Function]}
      type="radio"
    />
    <span
      className="algae-ui-radio-inner"
    />
  </span>
  <span
    className="algae-ui-radio-text"
  >
    radio
  </span>
</label>
`;
